### ISSUE TITLE
[go_router] Remove navigatorBuilder

### ISSUE THREAD

**Issue by johnpryan** [flutter/flutter#108145]
Once new route types are added to support nested routing (https://github.com/flutter/flutter/issues/108141) we can deprecate navigatorBuilder and recommend ShellRoute instead.

**Comment by ahmednfwela** [comment]
the issue is pointing to itself, I think the correct one is  https://github.com/flutter/flutter/issues/108141 ?

**Comment by chunhtai** [comment]
blocked on https://github.com/flutter/flutter/issues/108141

**Comment by chunhtai** [comment]
merged into feature branch. check https://github.com/flutter/flutter/issues/108044 for progress
